smithy-typescript-codegen/src/test/java/software/amazon/smithy/typescript/codegen/endpointsV2/EndpointsV2GeneratorTest.java

```diff
@@ -0,0 +1,85 @@
+
+public class EndpointsV2GeneratorTest {
+    @Test
+    public void containsTrailingSemicolon() {
+        testEndpoints("endpoints.smithy",
+                "      },\n" +
+                "      {\n" +
+                "        \"documentation\": \"Fallback when region is unset\",\n" +
+                "        \"conditions\": [\n" +
+                "        ],\n" +
+                "        \"error\": \"Region must be set to resolve a valid endpoint\",\n" +
+                "        \"type\": \"error\",\n" +
+                "      },\n" +
+                "    ],\n" +
+                "  }\n" +
+                ";\n");
+    }
+
+    @Test
+    public void containsExtraContextParameter() {
+        MockManifest manifest = testEndpoints("endpoints.smithy",
+                "      },\n" +
+                "      \"Stage\": {\n" +
+                "        \"type\": \"String\",\n" +
+                "        \"required\": true,\n" +
+                "        \"default\": \"production\",\n" +
+                "      },\n");
+
+        String contents = manifest.getFileString(CodegenUtils.SOURCE_FOLDER + "/endpoint/EndpointParameters.ts").get();
+
+        assertThat(contents, containsString(
+                "  return {\n" +
+                "    ...options,\n" +
+                "    region: options.region ?? \"us-east-1\",\n" +
+                "    stage: options.stage ?? \"production\",\n" +
+                "    defaultSigningName: \"\",\n" +
+                "  }\n"
+        ));
+    }
+
+    private MockManifest testEndpoints(String filename, String expectedType) {
+        MockManifest manifest = new MockManifest();
+        PluginContext context = PluginContext.builder()
+                .pluginClassLoader(getClass().getClassLoader())
+                .model(Model.assembler()
+                        .addImport(getClass().getResource(filename))
+                        .discoverModels()
+                        .assemble()
+                        .unwrap())
+                .fileManifest(manifest)
+                .settings(Node.objectNodeBuilder()
+                        .withMember("service", Node.from("smithy.example#Example"))
+                        .withMember("package", Node.from("example"))
+                        .withMember("packageVersion", Node.from("1.0.0"))
+                        .build())
+                .build();
+
+        new TypeScriptCodegenPlugin().execute(context);
+       
+        assertThat(manifest.hasFile(CodegenUtils.SOURCE_FOLDER + "/endpoint/EndpointParameters.ts"),
+                is(true));
+        assertThat(manifest.hasFile(CodegenUtils.SOURCE_FOLDER + "/endpoint/endpointResolver.ts"),
+                is(true));
+
+        String contents = manifest.getFileString(CodegenUtils.SOURCE_FOLDER + "/endpoint/ruleset.ts").get();
+
+        assertThat(contents, containsString("export const ruleSet: RuleSetObject"));
+        assertThat(contents, containsString(expectedType));
```

Comment:
Not sure what `expectedType` means here. I think it'll be easier to read the tests if this assertion is removed from `testEndpoints(String filename)` and instead each `@Test` method does its specific assertions.

Reply:
i have changed as suggested